src/unum/tests/tests.c

```diff
@@ -167,6 +169,11 @@ static int run_platform_tests(int test_num, char *test_num_str)
         case U_TEST_DNS:
             return test_dns();
 
+#ifdef FEATURE_GZIP_REQUESTS
```

Comment:
This ifdef should be disabling only the gzip requests code, here it's minizip code that should be available independently from the FEATURE_GZIP_REQUESTS. 
You can use weakly declared stub "test_zip()" function, keep it in minizip subsystem and only include the test when the subsystem is compiled in (conditionally at runtime basin on test_zip != NULL).

Reply:
Made test_zip as weak function and added a run time check. Also built on a platform where zlib is not included. No build errors.